Add Zenduty Integration (#1552)
DataDog/integrations-extras: zenduty/README.md

## Overview

> Use the Zenduty integration to send Datadog alerts to the right team, notify them as per on-call schedules, and help them remediate and resolve incidents with speed.

Reviewer: Can we clarify where the alerts are being sent? Are the Datadog alerts being sent within Zenduty or within Datadog?

Author: Done!

> Use the Datadog-Zenduty integration to send Datadog alerts to the right team, notify them as per on-call schedules and help them remediate and resolve incidents with speed.

Reviewer 2: ```suggestion
Use the Zenduty integration to send Datadog alerts to the right team, notify them as per on-call schedules, and help them remediate and resolve incidents with speed.
```

Connect Zenduty with Datadog in order to:

> - Trigger and resolve incidents, get alerts for them and track issues from Datadog
> - Deploy on-call schedules, escalation policies, incident playbooks, postmortems and detailed analytics
> - Use Alert Rules to custom route specific Datadog alerts to certain users or teams, write suppression rules, auto add notes, responders and incident tasks

Reviewer 2: ```suggestion
- Trigger and resolve incidents, get alerts for them, and track issues from Datadog
- Deploy on-call schedules, escalation policies, incident playbooks, postmortems, and detailed analytics
- Use Alert Rules to custom route specific Datadog alerts to certain users or teams, write suppression rules, and automatically add notes, responders, and incident tasks
```

## Setup

> ### Zenduty:

Reviewer 2: ```suggestion
### Zenduty
```

> ### Zenduty

Reviewer: ```suggestion
In Zenduty, follow the steps below:
```

Reviewer: It would be helpful to hyperlink Zenduty

Author: Done!

> Follow the following steps inside Zenduty:

Reviewer: ```suggestion
Follow the following steps within Zenduty:
```

> Follow the following steps inside Zenduty:

Reviewer: ```suggestion

```

> 1. Go to **'Teams'** and click on the team you want to add the integration to.
>  
> 2. Navigate to **'Services'**, create a new service or select an existing one.
>  
> 3. Go to **'Integrations'** and then **'Add New Integration'**. Give it a name and select the application **'Datadog'** from the dropdown menu.
> 
> 4. Go to **'Configure'** under your integrations and copy the Datadog webhook URL generated.

Reviewer 2: ```suggestion
1. Go to **Teams** and click on the team you want to add the integration to.
 
2. Navigate to **Services**. Create a new service or select an existing one.
 
3. Go to **Integrations** and then **Add New Integration**. Give the integration a name, and select the application **Datadog** from the dropdown menu.

4. Go to **Configure** under your integrations, and copy the generated Datadog webhook URL.
```

> ### Datadog:

Reviewer 2: ```suggestion
### Datadog
```

> ### Datadog

Reviewer: ```suggestion
### Follow the steps below within Datadog:
```

> ### Datadog

Reviewer: ```suggestion
### In Datadog, follow the steps below:
```

> 5. From the sidebar, go to **Integrations**. Search for **'Webhooks'** from this page, and click the add button.

Reviewer 2: ```suggestion
5. From the sidebar, go to **Integrations**. Search for **Webhooks** from this page, and click the add button.
```

> 5. From the sidebar, go to **Integrations**. Search for **Webhooks** from this page, and click the add button.

Reviewer: If you are referring to the webhook page in Datadog, it would be helpful to add a hyperlink: https://app.datadoghq.com/integrations/webhooks?search=webhook

Author: Added!

> 6. Scroll down, click on the **'+New'** button in the Webhooks section. Fill in the name, the webhook URL copied from Zenduty and paste the following JSON in the payload box:

Reviewer 2: ```suggestion
6. Scroll down, click on the **+New** button in the Webhooks section. Fill in the name, the webhook URL copied from Zenduty, and paste the following JSON in the payload box:
```

```
{
"alert_id": "$ALERT_ID",
"hostname":"$HOSTNAME",
"date_posix":"$DATE_POSIX",
"aggreg_key":"$AGGREG_KEY",
"title": "$EVENT_TITLE",
"alert_status":"$ALERT_STATUS",
"alert_transition":"$ALERT_TRANSITION",
"link":"$LINK",
"event_msg":"$TEXT_ONLY_MSG"
}
```

> 7. Click on **'Save'**. The Datadog-Zenduty integration is complete.

Reviewer 2: ```suggestion
7. Click on **Save**. The Datadog Zenduty integration setup is complete.
```

>     See the [Zenduty documentation][1] for more details and to get the most out of this integration.

Reviewer: ```suggestion
See the [Zenduty documentation][1] for more details and to get the most out of this integration.
```

Author: Removed trailing whitespaces

**Note**: Mention ```@zenduty``` as a channel under **Notify your team** in the Datadog monitor's configuration to get alerts through Zenduty when Datadog incidents are created or resolved.

## Data Collected
### Metrics

> Zenduty integration does not include any metrics.

Reviewer 2: ```suggestion
The Zenduty integration does not include any metrics.
```

### Events

> Triggered, acknowledged and resolved events are displayed in Zenduty's dashboard

Reviewer 2: ```suggestion
Triggered, acknowledged, and resolved events are displayed in Zenduty's dashboard
```

> Triggered, acknowledged, and resolved events are displayed in Zenduty's dashboard

Reviewer 2: ```suggestion
Triggered, acknowledged, and resolved events are displayed in Zenduty's dashboard.
```

### Service Checks

> Zenduty integration does not include any service checks.

Reviewer 2: ```suggestion
The Zenduty integration does not include any service checks.
```
